src/testflinger_device_connectors/devices/oemrecovery/oemrecovery.py

```diff
@@ -89,6 +91,68 @@ def provision(self):
         self._run_cmd_list(recovery_cmds)
         self.check_device_booted()
 
+    @timeout(dec_timeout=3600)
+    def check_boot_state(self):
+        """check boot state"""
+        try:
+            test_username = self.job_data.get("test_data", {}).get(
+                "test_username", "ubuntu"
+            )
+            test_password = self.job_data.get("test_data", {}).get(
+                "test_password", "ubuntu"
+            )
+        except AttributeError:
+            test_username = "ubuntu"
+            test_password = "ubuntu"
+
+        while True:
+            try:
+                image_type = (
+                    subprocess.check_output(
+                        (
+                            "sshpass -p {} ssh -o StrictHostKeyChecking=no "
+                            "-o UserKnownHostsFile=/dev/null "
+                            "{}@{} snap model | grep model"
+                        ).format(
+                            test_password,
+                            test_username,
+                            self.config["device_ip"],
+                        ),
+                        shell=True,
+                    )
+                ).decode()
+            except Exception:
```

Comment:
We should catch something more specific here, which is a little tricky. CalledProcessError might not be enough because the ssh session could timeout, but you also don't want to confuse TimeoutError with something that could get thrown by wrapt-timeout.

Reply:
Hi Paul
    Timeout Error thrown by wrapt-timeout. would  to caller, will not be handled in this function